src/include/stir/IO/SAFIRCListmodeInputFileFormat.h

```diff
@@ -77,43 +80,36 @@ class SAFIRCListmodeInputFileFormat : public InputFileFormat<CListModeData>, pub
 	//! Checks in binary data file for correct signature.
 	virtual bool can_read(const FileSignature& signature, std::istream& input ) const
 	{
-		return this->actual_can_read(signature, input);
+		return false; // cannot read from istream
 	}
 
 	//! Checks in binary data file for correct signature (can be either "SAFIR CListModeData" or "MUPET CListModeData").
 	virtual bool can_read( const FileSignature& signature, const std::string& filename) const
 	{
+		// Looking for the right key in the parameter file
+		std::ifstream par_file(filename.c_str());
+		std::string key;
+		std::getline(par_file, key, ':');
+		key.erase(std::remove_if(key.begin(), key.end(), is_whitespace), key.end());
+		if( !boost::iequals(key, std::string("CListModeDataSAFIRParameters")) ) { 
+			return false;
+		}
+
 		bool can_parse = actual_do_parsing(filename);
 		std::ifstream data_file(listmode_filename.c_str(), std::ios::binary);
```

Comment:
seems you should get out if parsing failed. otherwise who knows what you'll be reading.